x-pack/test/observability_ai_assistant_functional/tests/contextual_insights/index.spec.ts

```diff
@@ -0,0 +1,163 @@
+  const log = getService('log');
+  const apmSynthtraceEsClient = getService('apmSynthtraceEsClient');
+  const { common } = getPageObjects(['header', 'common']);
+
+  async function generateErrorData() {
+    const start = moment().subtract(5, 'minutes').valueOf();
+    const end = moment().valueOf();
+    const serviceName = 'opbeans-go';
+
+    const serviceInstance = apm
+      .service({ name: serviceName, environment: 'production', agentName: 'go' })
+      .instance('instance-a');
+
+    const interval = '1m';
+    const documents = [
+      timerange(start, end)
+        .interval(interval)
+        .rate(50)
+        .generator((timestamp) =>
+          serviceInstance
+            .transaction({ transactionName: 'GET /banana' })
+            .errors(
+              serviceInstance
+                .error({ message: 'Some exception', type: 'exception' })
+                .timestamp(timestamp)
+            )
+            .duration(10)
+            .timestamp(timestamp)
+            .failure()
+        ),
+    ];
+
+    await apmSynthtraceEsClient.index(documents);
+  }
+
+  async function createOpenAiConnector(proxy: LlmProxy) {
+    await supertest
+      .post('/api/actions/connector')
+      .set('kbn-xsrf', 'foo')
+      .send({
+        name: 'foo',
+        config: {
+          apiProvider: 'OpenAI',
+          apiUrl: `http://localhost:${proxy.getPort()}`,
+          defaultModel: 'gpt-4',
+        },
+        secrets: { apiKey: 'myApiKey' },
+        connector_type_id: '.gen-ai',
+      })
+      .expect(200);
+  }
+
+  async function deleteConnectors() {
+    const connectors = await supertest.get('/api/actions/connectors').expect(200);
+    const promises = connectors.body.map((connector: { id: string }) => {
+      return supertest
+        .delete(`/api/actions/connector/${connector.id}`)
+        .set('kbn-xsrf', 'foo')
+        .expect(204);
+    });
+
+    try {
+      await Promise.all(promises);
+    } catch (e) {
+      if (e?.response?.status === 404) {
+        return;
+      }
+
+      log.debug('Error deleting connector');
+      throw e;
+    }
+  }
+
+  async function navigateToError() {
+    await common.navigateToApp('apm', {
+      hash: `/services/opbeans-go/errors?rangeFrom=now-15m&rangeTo=now`,
+    });
+
+    await testSubjects.click('errorGroupId');
+  }
+
+  describe('Contextual insights for APM errors', () => {
+    let proxy: LlmProxy;
+
+    before(async () => {
+      await Promise.all([
+        deleteConnectors(), // cleanup previous connectors
+        apmSynthtraceEsClient.clean(), // cleanup previous synthtrace data
+        ui.auth.login(), // login
+      ]);
+
+      await generateErrorData();
+    });
+
+    describe('when there are no connectors', () => {
+      it('should not show the contextual insight component', async () => {
+        await navigateToError();
+        await retry.try(async () => {
+          await testSubjects.missingOrFail('obsAiAssistantInsightButton');
```

Comment:
we have page objects, should we use those?

Reply:
We can but in this case I am testing things outside AI Assistant pages.  This specifically is the the assistant button on the error details page in APM but it could be rendered anywhere. So we could have a page object called like:

```json
{
  contextualInsights: {
    button: 'obsAiAssistantInsightButton'
  }
}
```

It's a component rather than a page but might not matter